### PR TITLE
[bitnami/nginx-ingress-controller] Add TCP/UDP ports name prefix param

### PR DESCRIPTION
**Pull request by Zakhar-Petrov**
### Description of the change

Ability to set port name prefix for TCP/UDP in ingress controller service.

### Benefits

This is required as some clouds (e.g. Yandex Cloud) doesn't allow ports with names started with numbers.

### Possible drawbacks

None.

### Applicable issues

None.

### Additional information

The same changes from original kubernetes helm chart: kubernetes/ingress-nginx#8458.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/nginx-ingress-controller/README.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ As an alternative, you can use of the preset configurations for pod affinity, po
 | `dhParam`                              | A base64ed Diffie-Hellman parameter                                                                                                                | `""`                                       |
 | `tcp`                                  | TCP service key:value pairs                                                                                                                        | `{}`                                       |
 | `udp`                                  | UDP service key:value pairs                                                                                                                        | `{}`                                       |
+| `portNamePrefix`                       | Prefix for TCP and UDP ports names in ingress controller service                                                                                   | `""`                                       |
```
Comment (juan131) [inline bitnami/nginx-ingress-controller/README.md:161]: Both `tcp` and `udp` parameters allow you to use any prefix for the port names. For instance, using the values below as example:
```yaml
tcp:
  PREFIX-FOO: "5555"
udp:
  PREFIX-BAR: "6666"
```
The result is what you want, see:

```console
$ helm template nginx-ingress-controller oci://registry-1.docker.io/bitnamicharts/nginx-ingress-controller -f values.yaml -s templates/controller-service.yaml
(...)
spec:
  (...)
  ports:
    (...)
    - name: PREFIX-FOO-tcp
      port: PREFIX-FOO
      protocol: TCP
      targetPort: PREFIX-FOO-tcp
    - name: PREFIX-BAR-udp
      port: PREFIX-BAR
      protocol: UDP
      targetPort: PREFIX-BAR-udp

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/nginx-ingress-controller/README.md (hunk 1 above)
Comment (Zakhar-Petrov) [reply bitnami/nginx-ingress-controller/README.md:161]: There is a wrong port declaration in the example above, according to [ServicePort](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#serviceport-v1-core) specification, it must be an integer.

One more example for tcp/udp `key` usage where we can't use arbitrary string is `containerPort` in [controller-deployment.yaml](https://github.com/bitnami/charts/blob/a8ceaef42ee34b57733f7d301441ff8f2e5cd349/bitnami/nginx-ingress-controller/templates/controller-deployment.yaml):

```yaml
    spec:
      containers:
        - name: controller
...
          ports:
...
            {{- range $key, $value := .Values.tcp }}
            - name: "{{ $key }}-tcp"
              containerPort: {{ $key }}
              protocol: TCP
            {{- end }}
            {{- range $key, $value := .Values.udp }}
            - name: "{{ $key }}-udp"
              containerPort: {{ $key }}
              protocol: UDP
            {{- end }}
```

It's not obvious from the values documentation, but `key` inside tcp/udp params must be a valid port number.

Comment (juan131) [reply bitnami/nginx-ingress-controller/README.md:161]: That's clearly a bug. It should be:
```yaml
            {{- range $key, $value := .Values.tcp }}
            - name: "{{ $key }}-tcp"
              containerPort: {{ $value }}
              protocol: TCP
            {{- end }}
            {{- range $key, $value := .Values.udp }}
            - name: "{{ $key }}-udp"
              containerPort: {{ $value }}
              protocol: UDP
            {{- end }}
```

Comment (Zakhar-Petrov) [reply bitnami/nginx-ingress-controller/README.md:161]: I don't think so. `key` is an external port and `value` must be presented in the following format `<namespace/service name>:<service port/name>:[PROXY]:[PROXY]`. Please see https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services for more details.

Also I've found an example for tcp/udp params in https://github.com/bitnami/charts/pull/1396 (see [diff lines](https://github.com/bitnami/charts/pull/1396/files#diff-e0b453a4d4535124ed6f01a74890dc5cf5187e750a5bf00ed566ef814012c6beR514)):

```yaml
# TCP service key:value pairs
# Ref: https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx/examples/tcp
##
tcp: {}
#  8080: "default/example-tcp-svc:9000"

# UDP service key:value pairs
# Ref: https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx/examples/udp
##
udp: {}
#  53: "kube-system/kube-dns:53"
```

Comment (juan131) [reply bitnami/nginx-ingress-controller/README.md:161]: > `key` is an external port and `value` must be presented in the following format `<namespace/service name>:<service port/name>:[PROXY]:[PROXY]`. Please see https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services for more details.

You're completely right, I wasn't familiar with this feature. I can confirm this is the expected format and what we use in the ConfigMap(s) below:

- https://github.com/bitnami/charts/blob/main/bitnami/nginx-ingress-controller/templates/tcp-configmap.yaml
- https://github.com/bitnami/charts/blob/main/bitnami/nginx-ingress-controller/templates/udp-configmap.yaml

File: bitnami/nginx-ingress-controller/templates/controller-service.yaml [hunk 2]
```diff
@@ -74,21 +74,21 @@ spec:
       {{- end }}
     {{- end }}
     {{- range $key, $value := .Values.tcp }}
-    - name: {{ $key }}-tcp
+    - name: {{ if $.Values.portNamePrefix }}{{ $.Values.portNamePrefix }}-{{ end }}{{ $key }}-tcp
       port: {{ $key }}
       protocol: TCP
-      targetPort: {{ $key }}-tcp
+      targetPort: {{ if $.Values.portNamePrefix }}{{ $.Values.portNamePrefix }}-{{ end }}{{ $key }}-tcp
```
Comment (juan131) [inline bitnami/nginx-ingress-controller/templates/controller-service.yaml:80]: We could maintain the `targetPort` and container port name in controller's deployment/daemonset unchanged right? I mean, the restrictions you're mentioning only affects the service ports' names

Comment (juan131) [reply bitnami/nginx-ingress-controller/templates/controller-service.yaml:80]: Yes please, let's only apply the changes to service ports' names

Comment (Zakhar-Petrov) [reply bitnami/nginx-ingress-controller/templates/controller-service.yaml]: Done, please check it

**Pull request merged**